umapi_client/connection.py

```diff
@@ -310,6 +313,20 @@ def execute_multiple(self, actions, immediate=True):
         for a in actions:
             if len(a.commands) == 0:
                 if self.logger: self.logger.warning("Sending action with no commands: %s", a.frame)
+            # throttling part 1a: split commands with group assignments > self.throttle_groups
+            # if an action gets more than 10 commands per verb/group type combination, then it will
+            # be split below
+            for i, c in enumerate(a.commands):
+                for verb in ['add', 'remove']:
+                    subcommand = c.get(verb, {})
+                    if not subcommand:
+                        continue
+                    for group_type in ['product', 'usergroup', 'productConfiguration']:
```

Comment:
Hmmm.  Two issues:

- We don't actually care about the group type, do we?  They all have the same limits.  So you can just iterate the dictionary with `six.iteritems` and use whatever the key is in the split command.
- You shouldn't have a list of constants in the code.  There is an enumerator of all the group types in the functional module (where this code is going).  If it turns out you need to iterate the group types, use the enumerator; that's what it's for.

Reply:
I had trouble importing GroupTypes without running into a circular dependency error.  If I move the check code to functional that won't be an issue (and may be moot as per your first point).